BoostConfig.cmake.in

```diff
@@ -0,0 +1,21 @@
+set(_boost_deps @BOOST_FIND_PACKAGE@)
+
+foreach(dep ${_boost_deps})
+  find_package(${dep} QUIET)
+  if(${dep}_FOUND)
+    list(APPEND _boost_deps_found ${dep})
+  else()
+    list(APPEND _boost_deps_missing ${dep})
+  endif()
+endforeach()
+
+message(STATUS "Boost dependencies found: ${_boost_deps_found}")
```

Comment:
`find_package` already has mechanisms to handle config package dependencies, I think you should follow those procedures (if I understand what you're doing):

https://cmake.org/cmake/help/v3.6/manual/cmake-packages.7.html#id17